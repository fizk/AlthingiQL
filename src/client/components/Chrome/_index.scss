@import "../../config";

.chrome {
    //ref
}

.chrome__header {
    margin: auto;
    background-color: map_get($color-primary, dark);
    background-image: linear-gradient(to right, #973999, #f8598b, #f7bf00);
}

.chrome__header-container {
    max-width: 1024px;
    margin: auto;
}

.chrome__main {
    max-width: 1024px;
    margin: auto;
}

.chrome__footer {
    max-width: 1024px;
    margin: auto;
}

