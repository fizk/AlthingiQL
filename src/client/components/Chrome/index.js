import React from 'react';
import PropTypes from 'prop-types';
import {Column, Grid, Row} from "../../elements/Grid/index";

class Chrome extends React.Component {
    static propTypes = {};

    static defaultProps = {};

    render() {
        return (
            <div className="chrome">
                <header className="chrome__header">
                    <div className="chrome__header-container">
                        <Grid>
                            <Row>
                                <Column>
                                    <nav>
                                        <input type="search"/>
                                    </nav>
                                </Column>
                            </Row>
                        </Grid>
                    </div>
                </header>
                <main className="chrome__main">
                    {this.props.children}
                </main>
                <footer className="chrome__footer"></footer>
            </div>
        )
    }
}

export {Chrome}
