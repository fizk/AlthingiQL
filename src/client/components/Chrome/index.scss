@import "../../config";
@import "../../reset";

.chrome {
    max-width: 1280px;
    margin: auto;
}

.chrome__footer {
    padding: map_get($space, xxl);
}


@font-face {
    font-family: 'tt-webfont';
    src: url('/fonts/tt-regular-webfont.woff2') format('woff2');
    font-weight: normal;
}

@font-face {
    font-family: 'tt-webfont';
    src: url('/fonts/tt-medium-webfont.woff2') format('woff2');
    font-weight: bold;
}

body {
    font-family: 'tt-webfont', sans-serif;
    line-height: 1.25;
    background-color: #fafafa;
    font-size: 100%;
    -webkit-text-size-adjust: 100%;
    font-variant-ligatures: none;
    -webkit-font-variant-ligatures: none;
    text-rendering: optimizeLegibility;
    -moz-osx-font-smoothing: grayscale;
    font-smoothing: antialiased;
    -webkit-font-smoothing: antialiased;
    color: map_get($font-color, main);
}

a,
a:hover,
a:visited,
a:active {
    //color: #0000ee;
    text-decoration: inherit;
}
