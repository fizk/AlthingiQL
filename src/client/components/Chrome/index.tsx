import * as React from "react";
import Application from "../../elements/Application";
import { Column, Grid, Row } from "../../elements/Grid";
import './index.scss';

export default class Chrome extends React.Component<{}, {}> {

    render() {
        return (
            <Application>
                <div className="chrome">
                    <header className="chrome__header" role="banner">
                        <div className="chrome__header-container">
                            <Grid>
                                <Row>
                                    <Column>
                                        <nav>
                                            <input
                                                aria-label="AÃ°alleit"
                                                type="search"
                                            />
                                        </nav>
                                    </Column>
                                </Row>
                            </Grid>
                        </div>
                    </header>
                    <main className="chrome__main" role="main">
                        {this.props.children}
                    </main>
                    <footer className="chrome__footer">@loggjafarthing</footer>
                </div>
            </Application>
        );
    }
}
