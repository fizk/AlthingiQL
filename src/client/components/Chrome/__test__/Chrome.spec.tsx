import * as React from 'react';
import Chrome from '../index';
import renderer from 'react-test-renderer';
import {BrowserRouter as Router} from 'react-router-dom';

describe('<Chrome />', () => {
    test('default props', () => {
        const tree = renderer.create(
            <Router >
                <Chrome />
            </Router>,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
