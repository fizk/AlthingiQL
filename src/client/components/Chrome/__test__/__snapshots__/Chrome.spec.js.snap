// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chrome /> default props 1`] = `
<div
  className="application"
>
  <div
    className="chrome"
  >
    <header
      className="chrome__header"
    >
      <div
        className="chrome__header-container"
      >
        <div
          className="container-fluid"
        >
          <div
            className="row"
          >
            <div
              className="col"
            >
              <nav>
                <input
                  type="search"
                />
              </nav>
            </div>
          </div>
        </div>
      </div>
    </header>
    <main
      className="chrome__main"
    />
    <footer
      className="chrome__footer"
    >
      @loggjafarthing
    </footer>
  </div>
</div>
`;
